docs/misc/archival_data_recovery.md

```diff
@@ -0,0 +1,120 @@
+# Archival node - Recovery of missing data
+
+# Incident description
+
+In early 2024 there have been a few incidents on archival node storage. As result of these incidents archival node might have lost data from January to March.
+
+The practical effect of this issue is that requests querying the state of an account can fail to succeed, returning instead an internal server error.
+
+# Check if my node has been impacted
+
+The simplest way to check whether or not a node has suffered data loss is to run one of the following queries:
+
+*replace <RPC_URL> with the correct URL (example: http://localhost:3030)*
+
+```bash
+curl -X POST <RPC_URL> \
+        -H "Content-Type: application/json" \
+        -d '
+        { "id": "dontcare", "jsonrpc": "2.0", "method": "query", "params": { "account_id": "b001b461c65aca5968a0afab3302a5387d128178c99ff5b2592796963407560a", "block_id": 109913260, "request_type": "view_account" } }'
+```
+
+```bash
+curl -X POST <RPC_URL> \
+        -H "Content-Type: application/json" \
+        -d '
+        { "id": "dontcare", "jsonrpc": "2.0", "method": "query", "params": { "account_id": "token2.near", "block_id": 114580308, "request_type": "view_account" } }'
+```
+
+```bash
+curl -X POST <RPC_URL> \
+        -H "Content-Type: application/json" \
+        -d '
+        { "id": "dontcare", "jsonrpc": "2.0", "method": "query", "params": { "account_id": "timpanic.tg", "block_id": 115185110, "request_type": "view_account" } }'
+```
+
+```bash
+curl -X POST <RPC_URL> \
+        -H "Content-Type: application/json" \
+        -d '
+        { "id": "dontcare", "jsonrpc": "2.0", "method": "query", "params": { "account_id": "01.near", "block_id": 115514400, "request_type": "view_account" } }'
+```
+
+If for any of the above requests you get an error of the kind `MissingTrieValue` it means that the node presents the issue.
+
+# Remediation steps
+
+## Option A (recommended): download a new DB snapshot
+
+
+1. Stop `neard`
+2. Delete the existing `hot` and `cold` databases. Example assuming default configuration:
+    ```bash
+    rm -rf ~/.near/hot-data && rm -rf ~/.near/cold-data
+    ```
+
+
+3. Download an up-to-date snapshot, following this guide: [Storage snapshots](https://near-nodes.io/archival/split-storage-archival#S3%20migration)
+
+## Option B: manually run recovery commands
+
+Follow the instructions below if, for any reason, you prefer to perform the manual recovery steps on your node instead of downloading a new snapshot.
+
+**Requirements:**
+
+* `neard` must be stopped while recovering data
+* `cold` storage must be mounted on an SSD disk or better
+* The config `resharding_config.batch_delay` must be set to 0.
```

Comment:
Maybe mention that the last two steps should be undone after the recovery is finished?